katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,30 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  POSTGRES_DIR = '/var/lib/pgsql'
+
+  begin
+    available_space = `df -k --total --exclude-type=tmpfs --output=avail`.split("\n")[-1].to_i
+    postgres_size = File.directory?(POSTGRES_DIR) ? `du -k -s #{POSTGRES_DIR}`.split[0].to_i : 0
+    if available_space < postgres_size
+      $stderr.puts "The postgres upgrade requires at least #{postgres_size} KB of storage."
+      kafo.class.exit 1
```

Comment:
Since https://github.com/theforeman/kafo/commit/64810c08a0fe08d5ef76b20144ce709ee52e6f9f I think this can just be `exit` (and below as well):

```suggestion
      exit 1
```